cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -1028,14 +997,14 @@ private RunRecordMeta getCompletedRun(List<Field<?>> prefix, final String runId)
   private Map<ProgramRunId, RunRecordMeta> getNonCompleteRuns(@Nullable ProgramId programId, String recordType,
                                                               final long startTime, final long endTime, int limit,
                                                               Predicate<RunRecordMeta> filter)
-    throws IOException {
+    throws IOException, TableNotFoundException {
     Predicate<RunRecordMeta> valuePredicate = andPredicate(getTimeRangePredicate(startTime, endTime), filter);
     List<Field<?>> prefix = getRunRecordProgramPrefix(recordType, programId);
     return getProgramRunIdMap(Range.singleton(prefix), valuePredicate, null, limit);
   }
 
   /**
-   * Converts MDSkeys in the map to ProgramIDs
+   * Converts the range in the map to ProgramIds.
```

Comment:
this comment is still inaccurate. It does not just convert to program ids. It reads from the store and returns the actual run records.

Reply:
done